app/services/service_provider_seeder.rb

```diff
@@ -55,10 +56,16 @@ def write_service_provider?(config)
     return true if rails_env != 'production'
 
     restrict_env = config['restrict_to_deploy_env']
+    in_prod = deploy_env == 'prod'
+    in_sandbox = !%w[prod staging].include?(deploy_env)
+    in_staging = deploy_env == 'staging'
 
-    is_production_or_has_a_restriction = (deploy_env == 'prod' || restrict_env.present?)
+    return true if restrict_env == 'prod' && in_prod
```

Comment:
Please double check me here but I'm 99% that lines 63-68 are the same, functionally, as the previous code, with the addition of line 65 (`restrict_to_deploy_env` set to `sandbox`). I found the old code hard to read / reason about so even though this is more verbose I think it's a bit clearer.

Reply:
very minor, but my immediate thought when glancing at this was that it could be rewritten as a case statement. Either way works, though!

Reply:
Thanks! I generally like guard clauses over case statements for Ruby but I had the same thought myself 😄